Sources/PortalSwift/Core/Keychain/PortalKeychain.swift

```diff
@@ -9,7 +9,17 @@ import Foundation
 
 /// The main interface for Portal to securely store the client's signing share.
 public class PortalKeychain {
-  public static var metadata: PortalKeychainMetadata?
+  private var _metadata: PortalKeychainMetadata?
+  public var metadata: PortalKeychainMetadata? {
+    get async throws {
+      if let _metadata {
```

Comment:
will this caching be bad if we need the state to change based on when the wallet is created?